src/test.rs

```diff
@@ -1,25 +1,28 @@
 use super::*;
 use core::alloc::Layout;
-use std::mem::{align_of, size_of};
+use std::mem::{align_of, size_of, MaybeUninit};
 use std::prelude::v1::*;
 
 fn new_heap() -> Heap {
     const HEAP_SIZE: usize = 1000;
-    let heap_space = Box::into_raw(Box::new([0u8; HEAP_SIZE]));
+    let heap_space = Box::leak(Box::new([MaybeUninit::uninit(); HEAP_SIZE]));
+    let assumed_location = heap_space.as_ptr() as usize;
 
-    let heap = unsafe { Heap::new(heap_space as usize, HEAP_SIZE) };
-    assert!(heap.bottom == heap_space as usize);
+    let heap = Heap::with_memory(heap_space);
+    assert!(heap.bottom == assumed_location);
     assert!(heap.size == HEAP_SIZE);
     heap
 }
 
 fn new_max_heap() -> Heap {
     const HEAP_SIZE: usize = 1024;
     const HEAP_SIZE_MAX: usize = 2048;
-    let heap_space = Box::into_raw(Box::new([0u8; HEAP_SIZE_MAX]));
+    let heap_space = Box::leak(Box::new([MaybeUninit::<u8>::uninit(); HEAP_SIZE_MAX]));
+    let assumed_location = heap_space.as_ptr() as usize;
 
-    let heap = unsafe { Heap::new(heap_space as usize, HEAP_SIZE) };
-    assert!(heap.bottom == heap_space as usize);
+    // Unsafe so that we have provenance over the whole allocation.
+    let heap = unsafe { Heap::new(assumed_location, HEAP_SIZE) };
+    assert!(heap.bottom == assumed_location);
     assert!(heap.size == HEAP_SIZE);
     heap
 }
```

Comment:
Instead of updating these tests, how about creating two new tests? This way, we would still test the old functions.

Reply:
Those are the constructors for all of the current tests, not a single test. The first update removes an unsafe, I can revert the update to `new_max_heap`. I can add dedicated tests for the current functions though, right.